### PR TITLE
Make it more obvious when a test is expected to fail.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: tests/CMakeLists.txt [hunk 1]
```diff
@@ -98,25 +98,30 @@ foreach(file ${ALL_TESTS})
     set(TEST_CONFIGURATIONS "slow")
   endif()
 
+  set(TEST_NAME ${file})
```
Comment (kasperl) [inline tests/CMakeLists.txt:101]: Should this be TEST_EXPECTATION_NAME for consistency?

File: tests/image/CMakeLists.txt [hunk 2]
```diff
@@ -46,8 +46,13 @@ foreach(input ${INPUT})
   add_dependencies(check ${build_snap})
   add_dependencies(check_slow ${build_snap})
 
+  set(TEST_NAME ${input})
```
Comment (kasperl) [inline tests/image/CMakeLists.txt:49]: TEST_EXPECTATION_NAME?

File: tests/lock_file/CMakeLists.txt [hunk 3]
```diff
@@ -22,31 +22,41 @@ foreach(dir ${PACKAGE_LOCK_TESTS})
   set(test1 "${test_prefix}/test.toit")
 
   if (NOT "${test1}" IN_LIST TOIT_SKIP_TESTS)
+    set(TEST_NAME1 ${test1})
```
Comment (kasperl) [inline tests/lock_file/CMakeLists.txt:25]: TEST_EXPECTATION_NAME_1?

File: tests/lock_file/CMakeLists.txt [hunk 4]
```diff
@@ -22,31 +22,41 @@ foreach(dir ${PACKAGE_LOCK_TESTS})
...
 
     if("${test1}" IN_LIST TOIT_FAILING_TESTS)
-      set_tests_properties(${test1} PROPERTIES WILL_FAIL TRUE)
+      set_tests_properties(${TEST_NAME1} PROPERTIES WILL_FAIL TRUE)
     endif()
   endif()
 
   set(test2 "${test_prefix}")
   if (NOT "${test2}" IN_LIST TOIT_SKIP_TESTS)
+    set(TEST_NAME2 ${test2})
```
Comment (kasperl) [inline tests/lock_file/CMakeLists.txt:45]: TEST_EXPECTATION_NAME_2?

File: tests/lsp/CMakeLists.txt [hunk 5]
```diff
@@ -50,44 +50,49 @@ foreach(file ${ALL_TESTS})
     continue()
   endif()
 
+  set(TEST_NAME ${file})
```
Comment (kasperl) [inline tests/lsp/CMakeLists.txt:53]: TEST_EXPECTATION_NAME?

File: tests/optimizations/CMakeLists.txt [hunk 6]
```diff
@@ -22,15 +22,20 @@ foreach(file ${TOIT_OPTIMIZATION_TESTS})
     continue()
   endif()
 
+  set(TEST_NAME ${file})
```
Comment (kasperl) [inline tests/optimizations/CMakeLists.txt:25]: TEST_EXPECTATION_NAME?

File: tests/profiler/CMakeLists.txt [hunk 7]
```diff
@@ -27,13 +27,18 @@ foreach(profiler_test ${PROFILER_TESTS})
     continue()
   endif()
 
+  set(TEST_NAME ${profiler_test_name})
```
Comment (kasperl) [inline tests/profiler/CMakeLists.txt:30]: TEST_EXPECTATION_NAME?

File: tests/toitp/CMakeLists.txt [hunk 8]
```diff
@@ -90,26 +90,36 @@ foreach(file ${TOITP_TESTS})
 
   file(RELATIVE_PATH toitp_test_name ${TOIT_SDK_SOURCE_DIR} ${toitp_test})
   if (NOT "${toitp_test_name}" IN_LIST TOIT_SKIP_TESTS)
+    set(TEST_NAME "${toitp_test_name}")
```
Comment (kasperl) [inline tests/toitp/CMakeLists.txt:93]: TEST_EXPECTATION_NAME?

File: tests/type_propagation/CMakeLists.txt [hunk 9]
```diff
@@ -32,8 +32,13 @@ function(add_tests flags)
     string(REPLACE " " "__" test_name ${relative}${flags})
 
     if(NOT "${test_name}" IN_LIST TOIT_SKIP_TESTS)
+      set(test_expectation_name "${test_name}")
```
Comment (kasperl) [inline tests/type_propagation/CMakeLists.txt:35]: TEST_EXPECTATION_NAME?

File: tests/valgrind/CMakeLists.txt [hunk 10]
```diff
@@ -45,8 +45,12 @@ foreach(file ${VALGRIND_TESTS})
 
   set(valgrind_xml_prefix ${VALGRIND_TEST_DIR}/${file}.valgrind)
 
+  set(test_expectation_name "${test_name}")
```
Comment (kasperl) [inline tests/valgrind/CMakeLists.txt:48]: TEST_EXPECTATION_NAME?

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: tests/CMakeLists.txt (hunk 1 above)
Comment (floitsch) [reply tests/CMakeLists.txt]: done.

File: tests/image/CMakeLists.txt (hunk 2 above)
Comment (floitsch) [reply tests/image/CMakeLists.txt]: done.

File: tests/lock_file/CMakeLists.txt (hunk 3 above)
Comment (floitsch) [reply tests/lock_file/CMakeLists.txt]: done.

File: tests/lock_file/CMakeLists.txt (hunk 4 above)
Comment (floitsch) [reply tests/lock_file/CMakeLists.txt]: done.

File: tests/lsp/CMakeLists.txt (hunk 5 above)
Comment (floitsch) [reply tests/lsp/CMakeLists.txt]: done.

File: tests/optimizations/CMakeLists.txt (hunk 6 above)
Comment (floitsch) [reply tests/optimizations/CMakeLists.txt]: done.

File: tests/profiler/CMakeLists.txt (hunk 7 above)
Comment (floitsch) [reply tests/profiler/CMakeLists.txt]: done.

File: tests/toitp/CMakeLists.txt (hunk 8 above)
Comment (floitsch) [reply tests/toitp/CMakeLists.txt]: done.

File: tests/type_propagation/CMakeLists.txt (hunk 9 above)
Comment (floitsch) [reply tests/type_propagation/CMakeLists.txt]: done.

**Pull request merged**